### PR TITLE
[sgd] Avoid parameter "gotcha" for learning rate scheduler

### LINKED ISSUE #8103: [sgd] difference between RaySGD and PyTorch on single node for mnist
### What is your question?

*Ray version and other system information (Python version, TensorFlow version, OS):*
Python 3.8
Ray 0.9.0.dev0
PyTorch 1.4.0
Ubuntu 18.04
Tesla V100 with Driver Version: 440.33.01    CUDA Version: 10.2

Hi guys,

I'm trying to manually port an example from PyTorch repo to train a basic model on Mnist dataset: [link](https://github.com/pytorch/examples/blob/master/mnist/main.py)

I've done model configuration and training process identical, but getting strange results. 
Here you can see the original `main.py` and ported version `ray_main.py`: [gist](https://gist.github.com/PovelikinRostislav/b73362734d7c8e1f487dcc51bd5a7d10)

To launch them in identical manner:
```
python main.py --epochs 5
python ray_main.py --epochs 5 # assumed that local cluster is turned on and available by ray.init(addres='auto')
```

Here is the log of the `ray_main.py`. From epoch to epoch there is no changes in `val_accuracy` results despite the fact that `last_val_accuracy` is changing - why?
```
$ python ray_main.py --epochs 5
2020-04-20 14:14:17,105 WARNING worker.py:797 -- When connecting to an existing cluster, _internal_config must match the cluster's _internal_config.
{'num_samples': 60000, 'epoch': 1, 'batch_count': 938, 'train_loss': 1.4871701224009195, 'last_train_loss': 1.4667582511901855}
{'num_samples': 10000, 'batch_count': 10, 'val_loss': 1.3279935002326966, 'last_val_loss': 1.3232425451278687, 'val_accuracy': 0.6948, 'last_val_accuracy': 0.717}
==========
{'num_samples': 60000, 'epoch': 2, 'batch_count': 938, 'train_loss': 1.4828644436518352, 'last_train_loss': 1.473941683769226}
{'num_samples': 10000, 'batch_count': 10, 'val_loss': 1.327993595600128, 'last_val_loss': 1.3213943243026733, 'val_accuracy': 0.6948, 'last_val_accuracy': 0.696}
==========
{'num_samples': 60000, 'epoch': 3, 'batch_count': 938, 'train_loss': 1.4827248650868734, 'last_train_loss': 1.3438866138458252}
{'num_samples': 10000, 'batch_count': 10, 'val_loss': 1.3279935836791992, 'last_val_loss': 1.3088603019714355, 'val_accuracy': 0.6948, 'last_val_accuracy': 0.709}
==========
{'num_samples': 60000, 'epoch': 4, 'batch_count': 938, 'train_loss': 1.482667107327779, 'last_train_loss': 1.3714860677719116}
{'num_samples': 10000, 'batch_count': 10, 'val_loss': 1.3279935836791992, 'last_val_loss': 1.311557412147522, 'val_accuracy': 0.6948, 'last_val_accuracy': 0.709}
==========
{'num_samples': 60000, 'epoch': 5, 'batch_count': 938, 'train_loss': 1.4820500887552897, 'last_train_loss': 1.4103856086730957}
{'num_samples': 10000, 'batch_count': 10, 'val_loss': 1.3279935359954833, 'last_val_loss': 1.3126513957977295, 'val_accuracy': 0.6948, 'last_val_accuracy': 0.703}
==========
```

And to compare it with the results from the `main.py`. The difference in the accuracy is dramatic. 
```
$ python main.py --epochs 5 --log-interval 937
Train Epoch: 1 [0/60000 (0%)]   Loss: 2.333409
Train Epoch: 1 [29984/60000 (100%)]     Loss: 0.015671

Test set: Average loss: 0.0554, Accuracy: 9812/10000 (98%)

Train Epoch: 2 [0/60000 (0%)]   Loss: 0.038375
Train Epoch: 2 [29984/60000 (100%)]     Loss: 0.107407

Test set: Average loss: 0.0371, Accuracy: 9868/10000 (99%)

Train Epoch: 3 [0/60000 (0%)]   Loss: 0.017816
Train Epoch: 3 [29984/60000 (100%)]     Loss: 0.000814

Test set: Average loss: 0.0326, Accuracy: 9894/10000 (99%)

Train Epoch: 4 [0/60000 (0%)]   Loss: 0.002267
Train Epoch: 4 [29984/60000 (100%)]     Loss: 0.003287

Test set: Average loss: 0.0301, Accuracy: 9901/10000 (99%)

Train Epoch: 5 [0/60000 (0%)]   Loss: 0.045883
Train Epoch: 5 [29984/60000 (100%)]     Loss: 0.008944

Test set: Average loss: 0.0295, Accuracy: 9904/10000 (99%)
```

What's the reason? May I miss something?

### PR DESCRIPTION
**Pull request by richardliaw**
## Why are these changes needed?

More than 2 people have ran into this issue.

## Related issue number

Closes #8103 

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

**Review by edoakes — APPROVED** [review]
lgtm
